code/general/map_lookup_vs_pattern_matching.exs

```diff
@@ -12,22 +12,54 @@ defmodule Lookup.Slow do
   end
 end
 
-defmodule Lookup.Fast do
-  def int_for(str) do
-    do_int_for(str)
-  end
+defmodule Lookup.FastDidatic do
+  @moduledoc """
+  This is a didatic implementation which illustrates the code as the compiler sees it.
+  The downside is that if our key-value structure grows, the code becomes unwieldy.
+  """
+
+  # The base-clause can be declared without a do-block, which helps with readability.
+  def int_for(str)
```

Comment:
I took the liberty to refactor this to avoid the need for a secondary function. If you see fit, I can roll this change back